bfa-automation-java/src/main/java/com/bfa/core/page/Page.java

```diff
@@ -0,0 +1,65 @@
+package com.bfa.core.page;
+
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import static com.bfa.core.util.BfaConfig.getConfigValueAsInteger;
+import static com.bfa.core.util.LocatorUtils.getLocatorByKey;
+import static com.bfa.core.util.ThreadUtils.sleepCurrentThread;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
+
+/**
+ * Created by ninhdoan on 12/24/16.
+ */
+public class Page {
+
+   public static final String DEFAULT_WAIT_TIME_IN_SECONDS_KEY = "webdriver.defaultWaitTimeInSeconds";
+
+   private static final long DEFAULT_SLEEP_TIME_IN_MILLISECONDS = 100;
+
+   protected final WebDriver webDriver;
+
+   public Page(WebDriver webDriver) {
+      this.webDriver = webDriver;
+   }
+
+   public WebElement waitToGetVisibleWebElement(String webElementLocatorKey, long waitTimeInSeconds) {
+      WebDriverWait webDriverWait = new WebDriverWait(this.webDriver, waitTimeInSeconds);
+      WebElement visibleWebElement = webDriverWait.until(visibilityOfElementLocated(getLocatorByKey(webElementLocatorKey)));
+      return visibleWebElement;
+   }
+
+   public WebElement waitToGetVisibleWebElement(String webElementLocatorKey) {
+      return waitToGetVisibleWebElement(webElementLocatorKey, getConfigValueAsInteger(DEFAULT_WAIT_TIME_IN_SECONDS_KEY));
+   }
+
+   public String waitToGetNewUrl() {
+      return waitToGetNewUrl(getConfigValueAsInteger(DEFAULT_WAIT_TIME_IN_SECONDS_KEY));
+   }
+
+   public String waitToGetNewUrl(long waitTimeInSeconds) {
+      String currentUrl = webDriver.getCurrentUrl();
+      String newUrl = null;
+
+      long waitTimeInMilliseconds = waitTimeInSeconds * 1000;
+      long sleptTime = 0;
+      boolean isTheSameUrlAsBefore = false;
+
+      do {
+         sleepCurrentThread(DEFAULT_SLEEP_TIME_IN_MILLISECONDS);
+         sleptTime += DEFAULT_SLEEP_TIME_IN_MILLISECONDS;
+
+         newUrl = webDriver.getCurrentUrl();
+         isTheSameUrlAsBefore = newUrl == currentUrl || newUrl.equals(currentUrl);
+
+      } while (sleptTime < waitTimeInMilliseconds && isTheSameUrlAsBefore);
+
+      return newUrl;
+   }
```

Comment:
Please use ```ExpectedCondition``` instead of implement this wait. See [here](http://www.programcreek.com/java-api-examples/index.php?api=org.openqa.selenium.support.ui.ExpectedCondition) as example.